### PR TITLE
[feat] add recv function

### PR DESCRIPTION
**Pull request by monkingxue**
I found the `V8Worker2.recv` was called redundantly when I import `dispach.ts`. In fact, what we want is calling `recv` just once when deno init, so I modify the code.
I also add an util function named `once`, because I find other cases where function just be called once, such as `sub("start")`.

#### Checklist

- [x] `make test`  passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by monkingxue** [comment]
@qti3e Please review it, thx~

File: util.ts [hunk 1]
```diff
@@ -51,3 +51,19 @@ export function createResolvable<T>(): Resolvable<T> {
   });
   return Object.assign(promise, methods) as Resolvable<T>;
 }
+
+/* tslint:disable:no-any */
+export function once(fn: Function) {
```
Comment (qti3e) [inline util.ts]: I'd rather call this function `onceStrict`. (because it throws error when you call it more than once)

File: util.ts [hunk 2]
```diff
@@ -51,3 +51,19 @@ export function createResolvable<T>(): Resolvable<T> {
...
+  let haveCalled = false;
+  let result: any;
+
+  return (...args: any[]) => {
```
Comment (qti3e) [inline util.ts]: please don't use arrow function here so we can use later:
```js
const cb = onceStrict(fn);
cb.apply(something, ...);
```

File: util.ts [hunk 3]
```diff
@@ -51,3 +51,19 @@ export function createResolvable<T>(): Resolvable<T> {
...
+    if (haveCalled) {
+      throw Error("This function can only be called once");
+    }
+
+    haveCalled = true;
+    result = fn(...args);
```
Comment (qti3e) [inline util.ts]: following my previous comment, please change this line to:
```ts
result = fn.apply(this, args)
```

File: testdata/error.ts.out [hunk 4]
```diff
@@ -1,10 +1,12 @@
 /main.js:[WILDCARD]
-                throw _iteratorError;
-                ^
+                    throw _iteratorError;
+                    ^
```
[the remark this answers is not in the document]
Comment (monkingxue) [reply testdata/error.ts.out]: It's a good question but I don't know why, so sorry~

**Comment by monkingxue** [comment]
@ry Please review it, thx~

**Comment by ry** [comment]
@monkingxue As you know we're doing a major reorg and this patch is stale. Sorry about that (but that's life at the bleeding edge :)). Closing this now. I appreciate the contribution tho - hopefully things will settle down in a few days.

**Pull request closed without merging**